### PR TITLE
Form Control Enhancements

### PR DESCRIPTION
**Pull request by jbouder**
In order to simplify the creation of Forms, all applicable form controls should be updated to such that they can be used independently, while still providing a label, helper text, error messages, and validation status. This update supports that by adding all applicable props to the form group component, then wrapping each form control with the form group.

## Description

- Added Helper Text component
- Added all applicable props to form group
- Updated text input and text area with wrapper form group and related props
- Updated stories with new example usage
- Fix phone number pattern regex

## Related Issue

N/A

## Motivation and Context

- Simplify form component use

## How Has This Been Tested?

- Local Testing

## Screenshots (if appropriate):
Screenshot 2024-08-12 at 8 50 39 AM
Screenshot 2024-08-12 at 8 51 14 AM
Screenshot 2024-08-16 at 10 25 09 AM
Screenshot 2024-08-16 at 11 53 58 AM

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import globals from 'globals';
 import pluginJs from '@eslint/js';
-import stylisticTs from '@stylistic/eslint-plugin-ts';
 import tseslint from 'typescript-eslint';
+import prettierRecommended from 'eslint-plugin-prettier/recommended';
```
Comment (jbouder) [inline eslint.config.js:4]: This is unrelated. Realized this was conflicting with vscode settings, so went back to prettier

**Review by kildre — COMMENTED** [review]
Left one comment on the phone formatting

File: packages/comet-uswds/src/components/form/form.stories.tsx [hunk 2]
```diff
@@ -1,49 +1,127 @@
...
+      />
+      <TextInput
+        id="phone"
+        name="phone"
+        label="Phone Number"
+        helperText="Enter your phone number"
+        mask="phone_number"
+        value={phone}
+        onChange={(e) => setPhone(e.target.value)}
+      />
```
[the remark this answers is not in the document]
Comment (jbouder) [reply packages/comet-uswds/src/components/form/form.stories.tsx:105]: Thats actually on the todo list, but will be worked after these changes are in.

**Review by kildre — APPROVED** [review]
Comment addressed. 

Approved!
